src/management-system-v2/lib/iamComponents.tsx

```diff
@@ -0,0 +1,105 @@
+'use client';
+
+import { ComponentProps, FC, PropsWithChildren, ReactNode, useEffect, useMemo } from 'react';
+import { useRouter } from 'next/navigation';
+import { Route } from 'next';
+import {
+  ResourceActionType,
+  ResourceType,
+} from 'proceed-management-system/src/backend/server/iam/authorization/permissionHelpers';
+import { handleOauthCallback, useAuthStore } from './iam';
+import { App } from 'antd';
+
+let oneShot = true;
+export const AuthCallbackListener: FC = () => {
+  const oauthCallback = useAuthStore((store) => store.oauthCallback);
+  const router = useRouter();
+  const { message: messageApi } = App.useApp();
+
+  useEffect(() => {
+    if (oneShot && process.env.NEXT_PUBLIC_USE_AUTH && router) {
+      oneShot = false;
```

Comment:
This should probably be a `useRef()`, if you can't ensure that the effect is only run once otherwise.